### DISCUSSION TITLE
tasmota compiled gitpod-tasmocompiler, doesn't keep permanent setting data

### DISCUSSION THREAD

**Discussion by 3N37** [benzino77/tasmocompiler#123] [Q&A] [answered]
good morning everyone,
I have compiled with gitpod tasmota with the thermostat option.
I uploaded the .bin file on a wemos d1mini it works but it doesn't keep the setting at shutdown, I also set the rules (rule 1....) as the manual says but at the shutdown of the wemos all the configuration is lost.
I ask you if there is some trick to get the permanent setting.

**Comment by benzino77** [comment, marked as answer]
Hi there,
Did you compile the Wemos D1 mini version of Tasmota (in other words choose board version as "Wemos D1 mini" in the last step)?
Did you flash the image using Tasmotizer?

If answer to both question is "Yes", then it is a problem which arise once or twice.  Generally Tasmotizer "does not like" images build specifically for 4M boards (that is my observation). I don't know why, but it doesn't. In such case you have two options:
1. flash Tasmota to Wemos using other tool that Tasmotizer
2.  Compile "regular" image (choosing default board version: SonOff 1Mbit). It will not harm your wemos or make it unusable - it will work perfectly good, but in case you want to upgrade to new firmware version you will have to use compressed version of bin file or use two step procedure (with tasmota-minimal, as I remember, as an intermediate step) 

There is new version of TasmoCompiler. Not released yet. You can try it here: [gitpod](https://gitpod.io/#https://github.com/benzino77/tasmocompiler/tree/pa/xversion)

**Reply by 3N37** [reply]
magic benzino77  !!
I have replaced wemos 4mb with sonoff 1mb and finally saves the configuration!
Thanks.
I tried with firefox84.0.2 the new tasmocompiler in gitpod I noticed that it is a little slower than the current one, I get to a screen where it asks me to open a pop up, I grant the 'opening of the pop up and the page remains blocked with a black screen.

**Reply by benzino77** [reply]
Nice to hear it. 
Sometimes there is a problem with gitpod itself. Try in 5 minutes. 
New version of TC brings a bunch of new features.
